pkg/nfd-topology-updater/nfd-topology-updater.go

```diff
@@ -337,6 +339,46 @@ func (w *nfdTopologyUpdater) updateNodeResourceTopology(zoneInfo v1alpha2.ZoneLi
 	return nil
 }
 
+func readAttributeFromFile(entry string) (v1alpha2.AttributeInfo, error) {
+	cpus, err := os.ReadFile(entry)
+	if err != nil {
+		return v1alpha2.AttributeInfo{}, err
+	}
+
+	dir, _ := filepath.Split(entry)
+	kind := filepath.Base(dir)
+
+	attr := v1alpha2.AttributeInfo{
+		Name:  kind,
+		Value: strings.TrimSpace(string(cpus)),
+	}
+
+	return attr, nil
+}
+
+// Dicsover E/P cores
+func discoverCpuCores() v1alpha2.AttributeList {
+	attrList := v1alpha2.AttributeList{}
+
+	cpusPathGlob := hostpath.SysfsDir.Path("sys/devices/cpu_*/cpus")
+	cpuPaths, err := filepath.Glob(cpusPathGlob)
+	if err != nil {
+		klog.ErrorS(err, "error reading cpu entries", "cpusPathGlob", cpusPathGlob)
+		return attrList
+	}
+
+	for _, entry := range cpuPaths {
+		attr, err := readAttributeFromFile(entry)
+		if err != nil {
+			klog.ErrorS(err, "error reading cpu entry file", "entry", entry)
+		} else {
+			attrList = append(attrList, attr)
+		}
```

Comment:
Sorry @ozhuraki on the back'n'forth here (and a bit hasty review last round :roll_eyes: ) but looking at the latest iteration with the for loop I feel we could just ditch the `readAttributeFromFile()`. One additional reason being that getting the name of the attribute is a bit specialized (not take the name of the file but the name of the directory).

So one more suggestion:

```suggestion
		cpus, err := os.ReadFile(entry)
		if err != nil {
			klog.ErrorS(err, "error reading cpu entry file", "entry", entry)
		} else {
			attrList = append(attrList, v1alpha2.AttributeInfo{
				Name:  filepath.Base(filepath.Dir(entry)),
				Value: strings.TrimSpace(string(cpus)),
			}
		}
```

WDYT?